plugins/clnrest/utilities/rpc_routes.py

```diff
@@ -0,0 +1,71 @@
+import json5
+from flask import request, make_response, Response, stream_with_context
+from flask_restx import Namespace, Resource
+from .shared import call_rpc_method, verify_rune, process_help_response
+from .rpc_plugin import plugin
+
+methods_list = []
+rpcns = Namespace("RPCs")
+payload_model = rpcns.model("Payload", {}, None, False)
+
+@rpcns.route("/list-methods")
+class ListMethodsResource(Resource):
+    @rpcns.response(200, "Success")
+    @rpcns.response(500, "Server error")
+    def get(self):
+        """Get the list of all valid rpc methods"""
+        try:
+            help_response = call_rpc_method(plugin, "help", [])
+            html_content = process_help_response(help_response)
+            response = make_response(html_content)
+            response.headers["Content-Type"] = "text/html"
+            return response
+
+        except Exception as err:
+            plugin.log(f"Error: {err}", "error")
+            return json5.loads(str(err)), 500
+
+@rpcns.route("/<rpc_method>")
+class RpcMethodResource(Resource):
+    @rpcns.doc(security=[{"rune": [], "nodeid": []}])
+    @rpcns.doc(params={"rpc_method": (f"Name of the RPC method to be called")})
+    @rpcns.expect(payload_model, validate=False)
+    @rpcns.response(201, "Success")
+    @rpcns.response(500, "Server error")
+    def post(self, rpc_method):
+        """Call any valid core lightning method (check list-methods response)"""
+        try:
+            is_valid_rune = verify_rune(plugin, request)
+            
+            if "error" in is_valid_rune:
+                plugin.log(f"Error: {is_valid_rune}", "error")
+                raise Exception(is_valid_rune)
+
+        except Exception as err:
+            # Fix Me: Remove after lightningd checkrune is available
+            if not "unknown command" in str(err).lower():
+                return json5.loads(str(err)), 403
+        
+        try:
+            if request.is_json:
+                payload = request.get_json()
+            else:
+                payload = request.form.to_dict()
+            return call_rpc_method(plugin, rpc_method, payload), 201
+
+        except Exception as err:
+            plugin.log(f"Error: {err}", "error")
+            return json5.loads(str(err)), 500
+
+@rpcns.route("/notifications")
+class NotificationsResource(Resource):
```

Comment:
How do notifications work? Can you add an example app that uses this?

Reply:
The real question is, why do we not use a ws endpoint to get the notification? In this case, this plugin will blow up the ram memory because there is no limit on how many notifications you can insert into the queue

A possible solution to this (and I think the correct one) is that the client will subscribe to the notification (we can also think of filtering this notification by name and skip the `""` feature). e.g: https://docs.bmc.com/docs/display/public/btsim96/Using+the+Publish-Subscribe+REST+approach+to+receive+events

In coffee, we are implementing something similar with coffee to allow the user to get the notification back with a news stream https://docs.rs/actix-web-actors/latest/actix_web_actors/ws/index.html. It may be the solution also here.

To answer @niftynei I guess it is a pool API. When you call this API (you can see how to call it on the Open API page), you receive the notification of these call moments. But IMHO, this does not solve the problem from the UX point of view, is you are waiting for a notification, you are still called in a loop and match on the notification kind

Reply:
@niftynei I added an example in README. @vincenzopalazzo It is not a polling API and I was considering broadcasting queue Vs Websocket both as viable options for unidirectional communication. I picked queue first as it is lightweight and quicker option to implement. But with @cdecker's review, websocket becomes the obvious choice. So I will implement notifications with websocket server now.

Reply:
Resolving this conversation while keeping @cdecker's thread open for further updates.